File: albumloader/src/main/java/com/lee/album/entity/GalleryInfoEntity.java
package com.lee.album.entity;

import androidx.annotation.Nullable;

import java.io.Serializable;
import java.util.LinkedHashMap;
import java.util.List;

/**
 * Author : 李嘉伦
 * e-mail : lijialun@angogo.cn
 * desc   :
 */
public class GalleryInfoEntity implements Serializable {

    private String imgPath;
    private String imgName;
    private boolean selected;
    private String displayName;
    private String displayId;
    private int imgWidth;
    private int imgHeight;

    public int getImgWidth() {
        return imgWidth;
    }

    public GalleryInfoEntity setImgWidth(int imgWidth) {
        this.imgWidth = imgWidth;
        return this;
    }

    public int getImgHeight() {
        return imgHeight;
    }

    public GalleryInfoEntity setImgHeight(int imgHeight) {
        this.imgHeight = imgHeight;
        return this;
    }

    private LinkedHashMap<String, List<String>>size=new LinkedHashMap<>();

    public LinkedHashMap<String, List<String>> getSize() {
        return size;
    }

    public void setSize(LinkedHashMap<String, List<String>> size) {
        this.size = size;
    }






    public String getDisplayId() {
        return displayId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public GalleryInfoEntity setDisplayName(String displayName) {
        this.displayName = displayName;
        return this;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    public String getImgPath() {
        return imgPath;
    }

    public GalleryInfoEntity setImgPath(String imgPath) {
        this.imgPath = imgPath;
        return this;
    }

    public String getImgName() {
        return imgName;
    }

    public GalleryInfoEntity setImgName(String imgName) {
        this.imgName = imgName;
        return this;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        return this.imgPath.equals(((GalleryInfoEntity) obj).imgPath);
    }

    public GalleryInfoEntity setDisplayId(String id) {
        this.displayId = id;
        return this;
    }

}

File: albumloader/src/main/java/com/lee/album/entity/GalleryInfoEntity.kt
package com.lee.album.entity

import com.lee.album.entity.GalleryInfoEntity
import java.io.Serializable
import java.util.LinkedHashMap

/**
 * Author : 李嘉伦
 * e-mail : lijialun@angogo.cn
 * desc   :
 */
class GalleryInfoEntity : Serializable {
    var imgPath: String? = null
        private set
    var imgName: String? = null
        private set
    var isSelected = false
    var displayName: String? = null
        private set
    var displayId: String? = null
        private set
    var imgWidth = 0
        private set
    var imgHeight = 0
        private set

    fun setImgWidth(imgWidth: Int): GalleryInfoEntity {
        this.imgWidth = imgWidth
        return this
    }

    fun setImgHeight(imgHeight: Int): GalleryInfoEntity {
        this.imgHeight = imgHeight
        return this
    }

    var size = LinkedHashMap<String, List<String>>()
    fun setDisplayName(displayName: String?): GalleryInfoEntity {
        this.displayName = displayName
        return this
    }

    fun setImgPath(imgPath: String?): GalleryInfoEntity {
        this.imgPath = imgPath
        return this
    }

    fun setImgName(imgName: String?): GalleryInfoEntity {
        this.imgName = imgName
        return this
    }

    override fun equals(obj: Any?): Boolean {
        return imgPath == (obj as GalleryInfoEntity?)!!.imgPath
    }

    fun setDisplayId(id: String?): GalleryInfoEntity {
        displayId = id
        return this
    }
}